pkg/cache/cache_test.go

```diff
@@ -2120,3 +2120,235 @@ func messageOrEmpty(err error) string {
 	}
 	return err.Error()
 }
+
+func TestCache_AdmittedWorkloadsInLocalQueue(t *testing.T) {
+	type args struct {
+		localQueue *kueue.LocalQueue
+	}
+	tests := []struct {
+		name       string
+		args       args
+		want       int32
+		createFunc func() *Cache
+	}{
+		{
+			name: "test cluster queue ref not found",
+			args: args{
+				localQueue: &kueue.LocalQueue{
+					Spec: kueue.LocalQueueSpec{
+						ClusterQueue: "q1",
+					},
+				},
+			},
+			createFunc: func() *Cache {
+				return &Cache{
+					clusterQueues: map[string]*ClusterQueue{},
+				}
+			},
+			want: 0,
+		},
+		{
+			name: "test get admitted workloads in local queue",
+			args: args{
+				localQueue: &kueue.LocalQueue{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "q1",
+						Namespace: "ns1",
+					},
+					Spec: kueue.LocalQueueSpec{
+						ClusterQueue: "q1",
+					},
+				},
+			},
+			createFunc: func() *Cache {
+				return &Cache{
+					clusterQueues: map[string]*ClusterQueue{
+						"q1": {
+							Workloads: map[string]*workload.Info{
+								"app1": {
+									ClusterQueue: "q1",
+								},
+								"app2": {
+									ClusterQueue: "q2",
+								},
+							},
+							admittedWorkloadsPerQueue: map[string]int{
+								"ns1/q1": 1,
+							},
+						},
+					},
+				}
+			},
+			want: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := tt.createFunc()
+			if got := c.AdmittedWorkloadsInLocalQueue(tt.args.localQueue); got != tt.want {
```

Comment:
We have a combination test here: https://github.com/kubernetes-sigs/kueue/blob/b2a5e386d7e9c0e3346660dd01001734f631d7fd/pkg/cache/cache_test.go#L1523, but I think somehow this makes sense for `AdmittedWorkloadsInLocalQueue()` is now the only factor.